src/ServerRequest.php

```diff
@@ -166,7 +166,29 @@ private static function normalizeNestedFileSpec(array $files = [])
     public static function fromGlobals()
     {
         $method = isset($_SERVER['REQUEST_METHOD']) ? $_SERVER['REQUEST_METHOD'] : 'GET';
-        $headers = function_exists('getallheaders') ? getallheaders() : [];
+
+        $headers = [];
+        if (function_exists('getallheaders')) {
+            $headers = getallheaders();
+        } else {
+            foreach($_SERVER as $k => $v) {
+                if (substr($k, 0, 5) == 'HTTP_') {
+                    $k = explode('_', substr($k, 5));
+                    $header = [];
+                    foreach($k as $word) {
+                        $header[] = ucfirst(strtolower($word));
```

Comment:
This one lost, so reposting:

AFAIK headers are technically case-insensitive, so I'm not sure this is necessary. Also, in case of headers like `Content-MD5` or `X-Request-ID` it won't have the desired effect anyway.

But I would love to hear other opinions.